probe-rs/src/debug/debug_info.rs

```diff
@@ -328,15 +332,15 @@ impl DebugInfo {
         );
         function_root_variable.variable_node_type = VariableNodeType::DirectLookup;
         function_root_variable.name = VariableName::LocalScopeRoot;
-        function_variable_cache.cache_variable(None, function_root_variable, core)?;
+        function_variable_cache.cache_variable(None, function_root_variable, adapter)?;
         Ok(function_variable_cache)
     }
 
     /// This effects the on-demand expansion of lazy/deferred load of all the 'child' `Variable`s for a given 'parent'.
     pub fn cache_deferred_variables(
         &self,
         cache: &mut VariableCache,
-        core: &mut Core<'_>,
+        adapter: &mut impl MemoryInterface,
```

Comment:
Why is this not a `CoreInterface`?

Reply:
The code does not handle variables in registers yet afaik. Fixed it either way.

Reply:
Not sure about what you mean by "variables in registers", but unwind does use register values to unwind memory locations of variable values.